src/index.js

```diff
@@ -69,15 +96,34 @@ export default function ({ types: t }) {
           node.object = t.identifier(opts.systemGlobal);
         }
       },
-      Identifier({ node }) {
+      ReferencedIdentifier(path, state) {
+        if (path.node.name == 'define' &&
+          !path.scope.hasBinding('define') &&
+          (!t.isExpression(path.parentPath) ||
+            (t.isUnaryExpression(path.parentPath) && path.parentPath.node.operator === 'typeof'))) {
+          path.replaceWith(t.identifier('undefined'));
+        }
+      },
+      Identifier(path) {
+        let { node } = path;
         // test if file paths are used
         if (t.isIdentifier(node, { name: '__filename' }) ||
           t.isIdentifier(node, { name: '__dirname' })) {
           this.usesFilePaths = true;
         }
+
+        if (t.isIdentifier(node, { name: 'module' }) &&
+         t.isVariableDeclarator(path.parent)) {
+          this.moduleRedeclaredInOuterScope = true;
+        }
```

Comment:
This will check if `module` is declared *anywhere* in the file, but we should probably only care if it's defined in the *outer scope*. Maybe use something like `path.scope.hasBinding("module")` in `Program` visitor?